src/core/mod.rs

```diff
@@ -0,0 +1,39 @@
+mod game;
+mod menu;
+
+pub use bevy::prelude::*;
+
+use crate::ui;
+
+pub(super) fn plugin(app: &mut App) {
+    // Setup state
+    app.init_state::<CoreState>();
+    // Add state scoped entities for UI cleanup
+    app.enable_state_scoped_entities::<CoreState>();
+    // Print state transitions in debug builds
+    #[cfg(feature = "dev")]
+    app.add_systems(
+        Update,
+        bevy::dev_tools::states::log_transitions::<CoreState>,
+    );
+
+    // Setup, update, teardown
+    app.add_systems(Startup, setup);
+
+    // Sub plugins
+    app.add_plugins(ui::plugin);
```

Comment:
If `ui` is a sub plugin of `core`, why is it in a separate folder? But it's fine, I can bikeshed the directory structure later.

Reply:
not accurate comment